src/advanced_inv.h

```diff
@@ -391,10 +406,18 @@ class advanced_inventory
          * Add the item to the destination area.
          * @param destarea Where add the item to. This must not be AIM_ALL.
          * @param new_item The item to add.
-         * @param inv_item Pointer-pointer for the inventory's item pointer, if applicable.
-         * @return true if adding has been done, false if adding the item failed.
+         * @param count The amount to add items to add.
+         * @return Returns the amount of items that weren't addable, 0 if everything went fine.
+         */
+        int add_item( aim_location destarea, item &new_item, int count = 1);
+        /**
+         * Remove the item from source area. Must not be used on items with area
+         *      AIM_ALL or AIM_INVENTORY! (but is... and seems to work)
```

Comment:
Uhm, there are explicit asserts in that function.
```C++
    assert( sitem.area != AIM_ALL );        // should be a specific location instead
    assert( sitem.area != AIM_INVENTORY );  // does not work for inventory
```
The program *will* crash (per definition of `assert`) if this is called with an item in one of those areas.